lock-api/src/test/java/com/palantir/lock/client/LeasingTimelockClientTest.java

```diff
@@ -0,0 +1,222 @@
+import com.palantir.common.time.NanoTime;
+import com.palantir.lock.v2.IdentifiedTime;
+import com.palantir.lock.v2.LeasableLockResponse;
+import com.palantir.lock.v2.LeasableRefreshLockResponse;
+import com.palantir.lock.v2.LeasableStartIdentifiedAtlasDbTransactionResponse;
+import com.palantir.lock.v2.Lease;
+import com.palantir.lock.v2.LockImmutableTimestampResponse;
+import com.palantir.lock.v2.LockRequest;
+import com.palantir.lock.v2.LockResponse;
+import com.palantir.lock.v2.LockToken;
+import com.palantir.lock.v2.StartIdentifiedAtlasDbTransactionRequest;
+import com.palantir.lock.v2.StartIdentifiedAtlasDbTransactionResponse;
+import com.palantir.lock.v2.TimelockService;
+import com.palantir.lock.v2.TimestampAndPartition;
+
+public class LeasingTimelockClientTest {
+    private TimelockRpcClient timelockService = mock(TimelockRpcClient.class);
+    private TimelockService timelockClient;
+
+    private static final UUID LEADER_ID = UUID.randomUUID();
+
+    private static final LockToken LOCK_TOKEN = LockToken.of(UUID.randomUUID());
+    private static final LockResponse LOCK_RESPONSE = LockResponse.successful(LOCK_TOKEN);
+
+    @Before
+    public void setUp() {
+        when(timelockService.getLeaderTime()).thenReturn(IdentifiedTime.of(UUID.randomUUID(), NanoTime.now()));
+        timelockClient = LeasingTimelockClient.create(timelockService);
+    }
+
+    @Test
+    public void delegatesLockRequest() {
+        LockRequest lockRequest = mock(LockRequest.class);
+        when(timelockService.leasableLock(lockRequest)).thenReturn(
+                LeasableLockResponse.of(LOCK_RESPONSE, getLease()));
+
+        timelockClient.lock(lockRequest);
+
+        verify(timelockService).leasableLock(lockRequest);
+    }
+
+    @Test
+    public void lockResponeHasCorrectLeasedLock() {
+        LockRequest lockRequest = mock(LockRequest.class);
+        Lease lease = getLease();
+        when(timelockService.leasableLock(lockRequest)).thenReturn(
+                LeasableLockResponse.of(LOCK_RESPONSE, lease));
+
+        LockResponse clientResponse = timelockClient.lock(lockRequest);
+        LeasedLockToken leasedLockToken = (LeasedLockToken) clientResponse.getToken();
+        assertEquals(LOCK_TOKEN, leasedLockToken.serverToken());
+        assertEquals(lease, leasedLockToken.getLease());
+    }
+
+    @Test
+    public void delegatesStartAtlasdbTransactionRequest() {
+        StartIdentifiedAtlasDbTransactionRequest request = mock(StartIdentifiedAtlasDbTransactionRequest.class);
+        StartIdentifiedAtlasDbTransactionResponse response = startTransactionResponseWith(LOCK_TOKEN);
+        when(timelockService.leasableStartIdentifiedAtlasDbTransaction(request)).thenReturn(
+                LeasableStartIdentifiedAtlasDbTransactionResponse.of(response, getLease()));
+
+        timelockClient.startIdentifiedAtlasDbTransaction(request);
+
+        verify(timelockService).leasableStartIdentifiedAtlasDbTransaction(request);
```

Comment:
these kinds of verifications are largely fragile; they don't actually really test anything (you don't care that the method was called, you care if it returns the right response) and easily break.

In this case, you should be delegating that the result is as you expect, rather than verifying that the method was right